File: src/LeetCode56.cpp
class Solution {
public:
    vector<Interval> merge(vector<Interval>& intervals) {
        if (intervals.empty()) {
            return intervals;
        }
        vector<Interval> res;
        sort(intervals.begin(), intervals.end(),
             [](Interval a, Interval b){return a.start < b.start;});
        Interval pre = intervals[0];
        for (int i = 1; i < intervals.size(); i++) {
            Interval cur = intervals[i];
            if (cur.start > pre.end) {
                res.push_back(pre);
                pre = cur;
            } else {
                pre.end = max(pre.end, cur.end);
            }
        }
        res.push_back(pre);

        return res;
    }
};

File: src/LeetCode56.java
class Solution {
    public List<Interval> merge(List<Interval> intervals) {
        if (intervals == null || intervals.size() == 0) {
            return intervals;
        }

        Collections.sort(intervals, (a, b) -> a.start - b.start);
        List<Interval> res = new ArrayList<>();
        Interval prev = intervals.get(0);

        for (int i = 1; i < intervals.size(); i++) {
            Interval cur = intervals.get(i);
            if (cur.start > prev.end) {
                res.add(prev);
                prev = cur;
            } else {
                prev.end = Math.max(prev.end, cur.end);
            }
        }
        res.add(prev);
        return res;
    }
}
